src/io_threads.c

```diff
@@ -493,6 +493,8 @@ int tryOffloadFreeObjToIOThreads(robj *obj) {
 
     if (obj->refcount > 1) return C_ERR;
 
+    if (obj->type != OBJ_ENCODING_RAW || obj->type != OBJ_STRING) return C_ERR;
```

Comment:
In unstable, we can offload freeing of a nested object, like hashes or zsets.

This function is called only from dbSetValue, when a value is replaced, but there we actually want to keep the robj and only replace the robj->ptr if encoding is RAW. So there are not many scenarios where this will be used, I think.

It would be good if we can do that, without freeing the robj itself. We need the ptr and the encoding I think.

I have this draft code to add to dbReplaceValue:

```C
    /* Can we keep the object in the database and just replace the ptr? */
    if (old->refc == 1 && old->encoding == OBJ_ENCODING_RAW && val->encoding != OBJ_ENCODING_EMBSTR) {
        sdsfree(old->ptr); /* TODO: If it's very large, offload to io thread of lazy free. */
        old->ptr = val->ptr;
        old->type = val->type;
        old->encoding = val->encoding;
        decrRefCount(val);
        return old;
    }
```

I would like to keep the `old` robj for any encoding except EMBSTR. That is, if the key was for example a zset but it is replaced with a string value, we want to lazyfree the zset ptr but keep the robj.

Reply:
See #1337